source/extensions/filters/common/rbac/engine_impl.cc

```diff
@@ -33,17 +34,32 @@ bool RoleBasedAccessControlEngineImpl::allowed(const Network::Connection& connec
     }
   }
 
+  if (!matched && expr_ != nullptr) {
+    Protobuf::Arena arena;
+    auto eval_status = Expr::evaluate(*expr_, &arena, info, headers);
+
+    // evaluation error is effectively a denial
+    if (!eval_status.has_value()) {
+      return false;
```

Comment:
Errors are tricky to deal with in a policy. Are you sure you want to permit traffic based on the failure to evaluate a policy?

Reply:
I was thinking the failure of evaluation is considered not matched and will set `matched` to false. Whether or not to reject the traffic is depending on the `allowed_if_matched_`. So as long as the action is set to "ALLOW" (allow when matched), failure of evaluation is still going to reject a request. But if action is set to "DENY" (deny when matched), failure of evaluation is then going to permit the request because the condition is not matched.

Reply:
So if we only have a condition, and its fails to evaluate:
- for ALLOW, we reject, since we did not get "true" (aligning with IAM)
- for DENY, the question is whether we reject or not. I think we should reject, since that's the safe option, and the policy needs to be rewritten to be more fault tolerant. Basically, if you think of the condition as a program, we are throwing an exception.

Reply:
cc @TristonianJones since for a broader context.

Reply:
Failure to evaluate resolves to false in the match condition.